### PR TITLE
Make Gel foundations ESM usable

### PR DESCRIPTION
**Pull request by dr3**
Contributes to #557

**Overall change:** Adds ESM support to gel-foundations

**Code changes:**

- Uses ESM modules package key
- Restructures package to use nested `package.json`s to expose ESM code

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

**Comment by dr3** [comment]
Tested locally by publishing `gel-foundations` as `drew-testing-123@1.0.9` and importing into simorgh, string replacing `@bbc/gel-foundations` with `drew-testing-123`, Working as expected

File: packages/utilities/gel-foundations/index.test.jsx [hunk 1]
```diff
@@ -93,43 +82,39 @@ const scriptsExpectedExports = {
...
   korean: 'object',
   ethiopic: 'object',
-  default: 'object',
 };
 
 const expectedExports = {
   spacings: spacingsExpectedExports,
   breakpoints: breakpointsExpectedExports,
-  propTypes: propTypesExpectedExports,
+  'prop-types': propTypesExpectedExports,
```
Comment (pjlee11) [inline packages/utilities/gel-foundations/index.test.jsx:90]: 🤢

**Review by pjlee11 — APPROVED** [review]

**Comment by pjlee11** [comment]
Comparing `latest` to this branch `npm link`'ed into simorgh and seeing the top level gel-foundations smaller in the webpack bundle reports 🎉 

## Webpack report using ESM (linking this branch to simorgh)

image

## Webpack report not using ESM (latest)

image

**Review by chris-hinds — APPROVED** [review]

**Comment by dr3** [comment]
Doesnt need testing, will regression once published

**Pull request merged**